Automattic/wp-calypso
# Settings: Unify all section headers to use a single component

As we were working on site settings, we introduced several different helper methods for rendering a section header with a save button. These were doing the same thing in almost the same way, but were duplicating code. This PR introduces a universal component for that purpose, and updates all of site settings to use that component.

See #29887 and https://github.com/Automattic/wp-calypso/pull/29597#discussion_r244113122.

This PR should introduce no visual, behavioral or functional changes at all. 

#### Changes proposed in this Pull Request

* Settings: Introduce a `<SettingsSectionHeader />` component.
* Update all of site settings section headers to use that component for consistency.

#### Testing instructions

* Checkout this branch, or spin it up on calypso.live.
* Perform the following operations for:
  * A Jetpack site.
  * A WP.com site.
  * An Atomic site.
  * (bonus) a VIP site.
* Go to site settings of the site.
* Go through each of the settings sections and their respective cards.
* Make sure nothing has changed visually and everything continues to work like before (retrieving and saving any settings).

Fixes #29887.
